python/ray/data/_internal/execution/interfaces/execution_options.py

```diff
@@ -7,149 +7,189 @@
 from ray.util.annotations import DeveloperAPI
 
 
-@dataclass
 class ExecutionResources:
     """Specifies resources usage or resource limits for execution.
 
-    The value `None` represents unknown resource usage or an unspecified limit.
+    By default this class represents resource usage. Use `for_limits` or
+    set `default_to_inf` to True to create an object that represents resource limits.
     """
 
-    # CPU usage in cores (Ray logical CPU slots).
-    cpu: Optional[float] = None
+    def __init__(
+        self,
+        cpu: Optional[float] = None,
+        gpu: Optional[float] = None,
+        object_store_memory: Optional[float] = None,
+        default_to_inf: bool = False,
```

Comment:
I'm wondering about how we set `default_to_inf` to `False` by default.

If you previously did something like this, your code would break, right? Because then you're specifying all-zero resource limits?

```python
DataContext.get_current().execution_options.resource_limits = ExecutionResources()
```

Reply:
This won't break because we've overridden the setter method for `resource_limits`.
And the reason of defaulting to False is because there are fewer places that should default to False.